### PR TITLE
Implements 3 more shadow functions.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/client/shadow/w3dshadow.h [hunk 1]
```diff
@@ -110,6 +110,29 @@ class Shadow
         }
     }
 
+    void Set_Color(int color)
+    {
+        m_opacity = color & 0xFFFFFF;
+
+        if ((m_type & SHADOW_ALPHA_DECAL) != 0) {
+            m_color2 = (color << 24) | m_color1;
```
Comment (xezon) [inline src/game/client/shadow/w3dshadow.h]: What is going on here? Does the blue channel become alpha channel?

File: src/game/client/shadow/w3dshadow.h [hunk 2]
```diff
@@ -110,6 +110,29 @@ class Shadow
...
+    {
+        m_opacity = color & 0xFFFFFF;
+
+        if ((m_type & SHADOW_ALPHA_DECAL) != 0) {
+            m_color2 = (color << 24) | m_color1;
+        } else if ((m_type & SHADOW_ADDITIVE_DECAL) != 0) {
+            float o = m_opacity / 255.0f;
+            m_color2 = GameMath::Fast_To_Int_Truncate(((m_color1 >> 16) & 0xFF) * o)
+                | GameMath::Fast_To_Int_Truncate(((m_color1 >> 8) & 0xFF) * o)
+                | GameMath::Fast_To_Int_Truncate((m_color1 & 0xFF) * o);
```
Comment (xezon) [inline src/game/client/shadow/w3dshadow.h:123]: I have seen this pattern of color values somewhere else in Thyme. Is it possible that there could be a macro made for this?

Comment (xezon) [reply src/game/client/shadow/w3dshadow.h:123]: On second look, are you sure this logic is correct? It looks as if all color channels are pushed into the blue channel. Why?

Comment (tomsons26) [reply src/game/client/shadow/w3dshadow.h:123]: dunno if the logics are correct but pseudo matches

Comment (xezon) [reply src/game/client/shadow/w3dshadow.h:123]: Like so would make more sense if I am not mistaken.

Basically what happens is, color value is shifted to lowest 8 bits, masked and scaled by opacity, and then moved back to the byte position it belongs for R G B channel respectively.

```c++
            m_color2 = (GameMath::Fast_To_Int_Truncate(((m_color1 >> 16) & 0xFF) * o) << 16)
                | (GameMath::Fast_To_Int_Truncate(((m_color1 >> 8) & 0xFF) * o) << 8)
                | GameMath::Fast_To_Int_Truncate((m_color1 & 0xFF) * o);
```

**Pull request closed without merging**